modules/generator/processor/localblocks/query_range.go

```diff
@@ -62,15 +63,21 @@ func (p *Processor) QueryRange(ctx context.Context, req *tempopb.QueryRangeReque
 		jobErr = atomic.Error{}
 	)
 
+	maxSeries := int(req.MaxSeries)
+	totalRawResultsSeriesCount := int32(0)
```

Comment:
use the atomic int32 struct instead. it forces concurrent access to the var and is harder to make a mistake with. you can also use uber's atomic package which is more common in the codebase, but that matters less. 

perhaps one day we will standardize.